static/js/chart/draw.ts

```diff
@@ -661,9 +868,12 @@ function drawLineChart(
   const colorFn = getColorFn(legendText);
 
   let hasFilledInValues = false;
+  const timePoints = new Set<number>();
   for (const dataGroup of dataGroups) {
     const dataset = dataGroup.value.map((dp) => {
-      return [new Date(dp.label).getTime(), dp.value];
+      const time = new Date(dp.label).getTime();
```

Comment:
There are few places the label is converted to time. Would it help to add an optional "time" field in DataPoint struct and set it during initialization (and only line chart will set it)?

Reply:
I think the only problem is that since this field would be optional, when trying to access the time of a DataPoint, I would have to check that this field is set and if not, I would still want to try converting the label to time. Or can I just assume that if the time field is not set, then this datapoint is not for a time.

Reply:
Yeah, when building the DataPoint, should ensure "time" is set if the label is about time. DrawLineChart could assume upstream has set it, since we only draw time series in line chart.

Reply:
Then if we're assuming "time" is set if the label is about time, I should also update DrawGroupLineChart to use this time field and set it during initialization of the DataPoints for a group line chart? Maybe I should do this in a separate PR.

Reply:
separate PR sounds good